vmui: autocomplete usability improvements

Loori-R opened the pull request:
This pull request introduces improvements to the autocomplete feature in VMUI, addressing the issues identified in #5348. The key improvements are as follows:

- Changed suggestion loading approach.
- Implemented a `1000` request cache.
- Limited displayed suggestions to `50`.
- Removed hotkey for toggling autocomplete on/off, retained only "Quick Tip" (`Option/Ctrl + Space`).
- Fixed bug in 'start' and 'end' params for requests.

Loori-R commented on app/vmui/packages/vmui/src/hooks/useFetchQueryOptions.tsx:
```diff
@@ -5,140 +5,168 @@ import { AutocompleteOptions } from "../components/Main/Autocomplete/Autocomplet
-  const fetchData = async ({ urlSuffix, setter, type, params }: FetchDataArgs) => {
+  const fetchData = async ({ value, urlSuffix, setter, type, params }: FetchDataArgs) => {
+    abortControllerRef.current.abort();
+    abortControllerRef.current = new AbortController();
```
After calling `AbortController.signal.abort()`, the `AbortController` cannot be reused, so a new one must be created. 
In this case, the `AbortController` is needed to aborted a request that is no longer relevant to the user because they now require different values for suggestions.

Loori-R commented on app/vmui/packages/vmui/src/components/Configurators/QueryEditor/QueryEditorAutocomplete.tsx:
```diff
@@ -113,16 +120,23 @@ const QueryEditorAutocomplete: FC<QueryEditorAutocompleteProps> = ({
+        value={valueByContext}
+        options={options?.length < AUTOCOMPLETE_LIMITS.queryLimit ? options : []}
+        anchor={anchorEl}
+        minLength={context === QueryContextType.metricsql ? 2 : 0}
```
updated

dmitryk-dk reviewed (approved):
LGTM!

valyala commented:
FYI, I fixed VictoriaMetrics queries used for auto-complete at the commit ca7b48fc4b82ca0ad541579d30bfc56ee489aa28 . The following should be fixed:

- It would be great from readability PoV to rename `values` to `labelValues` inside `app/vmui/packages/vmui/src/hooks/useFetchQueryOptions.tsx`
- The `alt+shift+a` hotkey for auto-complete suggestion is hard to type. It would be great replacing it with `ctrl+space` hotkey.
- The hotkey sometimes doesn't trigger to auto-complete suggestion. Also the auto-complete sometimes stops working and isn't recovered after that. This should be fixed.
- It would be great allowing `:` and `.` chars in metric names and label names, e.g. auto-complete should work when the query input field contains `foo.bar`, `foo:bar`, `foo{a.b` or `foo{a:b`.

Loori-R commented:
Fixed the issues:

Changed the hotkey to `ctrl/option + space`
Fixed the bug where suggestions were not displaying
Expanded the character set in names
Renamed constants from `values` to `labelValues`

valyala commented:
@Loori-R , thanks for the pull request!
